specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1120 @@
+    resources: [<resource kind >, <resource kind>/status]
+    verbs: ["get", "list", "watch"]
+```
+
+This configuration advice SHALL NOT indicate a requirement for Kubernetes RBAC
+support.
+
+<!-- TODO: define aggregated API roles
+
+Ref:
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-channelable-manipulator-clusterrole
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-addressable-resolver-clusterrole
+-
+-->
+
+## Resource Overview
+
+The Knative Eventing API provides a set of primitives to support both
+point-to-point communication channels (`messaging.knative.dev`) and
+content-based event routing (`eventing.knative.dev`). This specification
+describes API interfaces of Knative Eventing resources as well as the supported
+[event routing](./data-plane.md) logic and configuration settings.
+
+At the moment, the Knative Eventing specification does not contemplate any
+non-Kubernetes-backed implementations, and therefore does not specifically
+define the mapping of kubernetes verbs (read, watch, patch, etc) to developer
+roles. See the [Overview documentation](./overview.md) for general definitions
+of the different API objects.
+
+## Error Signalling
+
+<!-- copied from ../serving/knative-api-specification-1.0.md#error-signalling -->
+
+The Knative API uses the
+[Kubernetes Conditions convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)
+to communicate errors and problems to the user. Each user-visible resource
+described in Resource Overview MUST have a `conditions` field in `status`, which
+must be a list of `Condition` objects of the following form (note that the
+actual API object types may be named `FooCondition` to allow better code
+generation and disambiguation between similar fields in the same `apiGroup`):
+
+<table>
+  <tr>
+   <td><strong>Field</strong>
+   </td>
+   <td><strong>Type</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+   <td><strong>Default Value</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><code>type</code>
+   </td>
+   <td><code>string</code>
+   </td>
+   <td>The category of the condition, as a short, CamelCase word or phrase.
+<p>
+This is the primary key of the Conditions list when viewed as a map.
+   </td>
+   <td>REQUIRED – No default
+   </td>
+  </tr>
+  <tr>
+   <td><code>status</code>
+   </td>
+   <td>Enum:<ul>
+
+<li>"True"
+<li>"False"
+<li>"Unknown"</li></ul>
+
+   </td>
+   <td>The last measured status of this condition.
+   </td>
+   <td>"Unknown"
+   </td>
+  </tr>
+  <tr>
+   <td><code>reason</code>
+   </td>
+   <td>string
+   </td>
+   <td>One-word CamelCase reason for the condition's last transition.
+   </td>
+   <td>""
+   </td>
+  </tr>
+  <tr>
+   <td><code>message</code>
+   </td>
+   <td>string
+   </td>
+   <td>Human-readable sentence describing the last transition.
+   </td>
+   <td>""
```

Comment:
is the default "" or is it "absent" ?  Same for the others. I would expect an OPTIONAL field to be missing if not set. Although, if we want, we can add text that says absent and "" are semantically equivalent.

Reply:
It looks like the recommended Golang json format is `omitempty`, so the default is omitted.

Reply:
The text on line 11 of `error-signalling.md` should clarify this; I think we could omit this and just have required/optional except that the default value of `status` is `Unknown` if it is omitted, to have only 3 states for that field.